app/controllers/recipes_controller.rb

```diff
@@ -13,10 +13,30 @@ def new
     @recipe = Recipe.new
   end
 
+  def create
+    @recipe = Recipe.new(recipe_params)
+    @recipe.save
+
+    redirect_to recipes_path
+  end
+
   def edit
     @recipe = Recipe.find(params[:id])
   end
 
+  def update
+    @recipe = Recipe.find(params[:id])
+    @recipe.update(recipe_params)
+    redirect_to recipe_path
```

Comment:
I think we need to add `@recipe` to the `recipe_path` so it know which recipe to go to.